core/src/main/java/org/apache/calcite/sql/SqlNumericLiteral.java

```diff
@@ -103,7 +103,7 @@ public boolean isExact() {
           BigDecimal bd = getValueNonNull();
           SqlTypeName result;
           // Will throw if the number cannot be represented as a long.
-          long l = bd.longValue();
+          long l = bd.longValueExact();
```

Comment:
As per the [documentation](https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html#longValueExact--), this will throw an `ArithmeticException` in case of overflow, should we catch it and fallback do decimal?

Reply:
yes, that is exactly what is going on here. This is within a try-catch block